components/kyma-environment-broker/internal/hyperscaler/azure/testing/testing.go

```diff
@@ -0,0 +1,165 @@
+package testing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/eventhub/mgmt/2017-04-01/eventhub"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/ptr"
+)
+
+// ensure the fake Client is implementing the interface
+var _ azure.AzureInterface = (*FakeNamespaceClient)(nil)
+
+/// A fake Client for Azure EventHubs Namespace handling
+type FakeNamespaceClient struct {
+	PersistEventhubsNamespaceError error
+	ResourceGroupError             error
+	AccessKeysError                error
+	AccessKeys                     *eventhub.AccessKeys
+	Tags                           azure.Tags
+	GetResourceGroupError          error
+	GetResourceGroupReturnValue    resources.Group
+	DeleteResourceGroupCalled      bool
+	DeleteResourceGroupError       error
+}
+
+func (nc *FakeNamespaceClient) GetEventhubAccessKeys(ctx context.Context, resourceGroupName string, namespaceName string, authorizationRuleName string) (result eventhub.AccessKeys, err error) {
+	if nc.AccessKeys != nil {
+		return *nc.AccessKeys, nil
+	}
+	return eventhub.AccessKeys{
+		PrimaryConnectionString: ptr.String("Endpoint=sb://name/;"),
+	}, nc.AccessKeysError
+}
+
+func (nc *FakeNamespaceClient) CreateResourceGroup(ctx context.Context, config *azure.Config, name string, tags azure.Tags) (resources.Group, error) {
+	nc.Tags = tags
+	return resources.Group{
+		Name: ptr.String("my-resourcegroup"),
+	}, nc.ResourceGroupError
+}
+
+func (nc *FakeNamespaceClient) GetResourceGroup(ctx context.Context, tags azure.Tags) (resources.Group, error) {
```

Comment:
(nit) If the function arguments are not used in its body, you can safely omit their names, and it shall read `GetResourceGroup(context.Context, azure.Tags)`.

Same comment applies for the rest of the fake client functions.